lib/src/test/java/vmath/Mat2Test.java

```diff
@@ -22,6 +22,33 @@ void givenIndices_whenGetCalled_thenReturnValueMatches() {
         //TODO
     }
 
+    @Test
+    void givenArgs_whenMat2Created_thenMatrixHasGivenValues() {
+        float[] validValues = { 2.0f, 3.0f, 4.0f, 5.0f };
+
+        Mat2 mat2 = new Mat2(validValues);
+
+        for (int i = 0; i < 2; i++) {
+            for (int j = 0; j < 2; j++) {
+                assertEquals(validValues[i * 2 + j], mat2.get(i, j));
+            }
+        }
+    }
+
+    @Test
+    void givenWrongSize_whenMat2Created_thenThrowsException() {
+
+        float[] invalidSize= { 2.0f, 3.0f, 4.0f};
+        assertThrows(IllegalArgumentException.class, () -> new Mat2(invalidSize));
+    }
+
+    @Test
+    void givenNaNValues_whenMat2Created_thenThrowsException() {
+        
+        float[] invalidValues = {Float.NaN,Float.NaN,Float.NaN, Float.NaN };
+        assertThrows(ArithmeticException.class, () -> new Mat2(invalidValues));
+    }
```

Comment:
Please remove the empty line 47 and place your "float[] invalidValues = {Float.NaN,Float.NaN,Float.NaN, Float.NaN };" there. However, add an empty line before "assertThrows(ArithmeticException.class, () -> new Mat2(invalidValues));". These changes must be made to maintain consistency with the already merged code.